gateway/src/main/java/io/camunda/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java

```diff
@@ -74,21 +72,140 @@ public void accept(final ActorControl actor) {
+   * |                 |      |                                                              +------------------------------+                               |                        |
+   * |                 |      |                                                                                                                             |                        |
+   * |                 |------|-----------                                                                                                                  |                        |
+   * +-----------------+      |           \-------------------------------------                                                         +------------------v-----+                  |
+   *                          |                                         Error   \-------------------------------------                   |                        |                  |
+   *                          |                                                                                       \-------------------   Resource Exhausted   |                  |
+   *                          |                                                                                                          |                        |                  |
+   *                          |                                                                                                          +------------------------+                  |
+   *                          |                                                                                                                                                      |
+   *                          |                                                                                                                                                      |
+   *                          |                                                                                                                                                      |
+   *                          +------------------------------------------------------------------------------------------------------------------------------------------------------+
+   *
+   * https://textik.com/#a2725e317ed87a9d
+   * @param request The request to handle
+   * @param responseObserver The stream to write the responses to
+   */
   @Override
   public void activateJobs(
       final ActivateJobsRequest request,
       final ServerStreamObserver<ActivateJobsResponse> responseObserver) {
-    final var longPollingRequest = toInflightActivateJobsRequest(request, responseObserver);
-    activateJobs(longPollingRequest);
+    actor.run(
+        () -> {
+          final var longPollingRequest =
+              new InflightActivateJobsRequest(
+                  ACTIVATE_JOBS_REQUEST_ID_GENERATOR.getAndIncrement(), request, responseObserver);
```

Comment:
:wrench: If `ACTIVATE_JOBS_REQUEST_ID_GENERATOR` is only accessed through the actor, we don't need to make it an atomic.